### PR TITLE
Story/csi 1191 iscsi login deployment

### PR DESCRIPTION
**Pull request by igorgonibm**
Deploy iscsi login

### CODE REVIEW

File: node/scripts/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#! /bin/bash -x
+
+echo "Install iscsid..."
+yum install -y iscsi-initiator-utils && yum clean all
```
Comment (oriyarde) [inline node/scripts/entrypoint.sh]: I don't think worker node should require internet access. this should be at compilation time.

File: node/scripts/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+echo "Starting iscsid..."
+iscsid -f &
+
+# sleep one second to make sure iscsid is running in background
+sleep 1
+
+echo "Starting driver with arguments $@..."
```
Comment (oriyarde) [inline node/scripts/entrypoint.sh]: what arguments?

File: node/scripts/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#! /bin/bash -x
+
+echo "Install iscsid..."
+yum install -y iscsi-initiator-utils && yum clean all
+
+echo "Starting iscsid..."
+iscsid -f &
+
+# sleep one second to make sure iscsid is running in background
```
Comment (oriyarde) [inline node/scripts/entrypoint.sh]: I would remove the sleep

File: Dockerfile-csi-node [hunk 4]
```diff
@@ -45,15 +45,19 @@ COPY --from=builder /go/src/github.com/ibm/ibm-block-csi-driver/common/config.ya
...
 COPY ./LICENSE /licenses/
 COPY ./NOTICES /licenses/
+COPY ./node/scripts/entrypoint.sh .
+RUN chmod 755 ./entrypoint.sh
 
 RUN mkdir /chroot
 ADD chroot-host-wrapper.sh /chroot
 RUN chmod 777 /chroot/chroot-host-wrapper.sh
 RUN    ln -s /chroot/chroot-host-wrapper.sh /chroot/blkid \
+    && ln -s /chroot/chroot-host-wrapper.sh /chroot/yum \
```
Comment (oriyarde) [inline Dockerfile-csi-node]: remove

File: Dockerfile-csi-node [hunk 5]
```diff
@@ -65,5 +69,4 @@ RUN    ln -s /chroot/chroot-host-wrapper.sh /chroot/blkid \
 
 ENV PATH="/chroot:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 
-
-ENTRYPOINT ["/root/ibm-block-csi-node-driver"]
+ENTRYPOINT ["./entrypoint.sh"]
```
Comment (oriyarde) [inline Dockerfile-csi-node]: use full path

File: node/scripts/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#! /bin/bash -x
```
Comment (oriyarde) [inline node/scripts/entrypoint.sh]: why is there a space between the hashbang and the `/bin/bash`?, and why the `-x`?

File: Dockerfile-csi-node [hunk 7]
```diff
@@ -45,15 +45,19 @@ COPY --from=builder /go/src/github.com/ibm/ibm-block-csi-driver/common/config.ya
...
 RUN mkdir /chroot
 ADD chroot-host-wrapper.sh /chroot
 RUN chmod 777 /chroot/chroot-host-wrapper.sh
 RUN    ln -s /chroot/chroot-host-wrapper.sh /chroot/blkid \
+    && ln -s /chroot/chroot-host-wrapper.sh /chroot/yum \
     && ln -s /chroot/chroot-host-wrapper.sh /chroot/blockdev \
     && ln -s /chroot/chroot-host-wrapper.sh /chroot/iscsiadm \
     && ln -s /chroot/chroot-host-wrapper.sh /chroot/lsblk \
     && ln -s /chroot/chroot-host-wrapper.sh /chroot/lsscsi \
+    && ln -s /chroot/chroot-host-wrapper.sh /chroot/iscsid \
```
Comment (oriyarde) [inline Dockerfile-csi-node]: move it right after `iscsiadm`

File: node/scripts/entrypoint.sh (hunk 2 above)
Comment (igorgonibm) [reply node/scripts/entrypoint.sh]: removed echo

File: Dockerfile-csi-node (hunk 4 above)
Comment (igorgonibm) [reply Dockerfile-csi-node]: removed

File: Dockerfile-csi-node (hunk 7 above)
Comment (igorgonibm) [reply Dockerfile-csi-node]: v

File: node/scripts/entrypoint.sh (hunk 1 above)
Comment (igorgonibm) [reply node/scripts/entrypoint.sh]: removed

File: node/scripts/entrypoint.sh (hunk 6 above)
Comment (igorgonibm) [reply node/scripts/entrypoint.sh]: v

File: Dockerfile-csi-node (hunk 7 above)
Comment (igorgonibm) [reply Dockerfile-csi-node]: v

File: node/scripts/entrypoint.sh (hunk 2 above)
Comment (igorgonibm) [reply node/scripts/entrypoint.sh]: removed

File: node/scripts/entrypoint.sh (hunk 6 above)
Comment (igorgonibm) [reply node/scripts/entrypoint.sh]: v

File: Dockerfile-csi-node (hunk 5 above)
Comment (igorgonibm) [reply Dockerfile-csi-node]: v

File: Dockerfile-csi-node [hunk 8]
```diff
@@ -45,25 +45,27 @@ COPY --from=builder /go/src/github.com/ibm/ibm-block-csi-driver/common/config.ya
...
+COPY ./node/scripts/entrypoint.sh .
+RUN chmod 755 ./entrypoint.sh
 
 RUN mkdir /chroot
 ADD chroot-host-wrapper.sh /chroot
 RUN chmod 777 /chroot/chroot-host-wrapper.sh
 RUN    ln -s /chroot/chroot-host-wrapper.sh /chroot/blkid \
     && ln -s /chroot/chroot-host-wrapper.sh /chroot/blockdev \
     && ln -s /chroot/chroot-host-wrapper.sh /chroot/iscsiadm \
+    && ln -s /chroot/chroot-host-wrapper.sh /chroot/iscsid \
```
Comment (27149chen) [inline Dockerfile-csi-node]: so you are using iscsid on node, right?

File: node/scripts/entrypoint.sh [hunk 9]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+/root/ibm-block-csi-node-driver $@
```
Comment (27149chen) [inline node/scripts/entrypoint.sh]: What's this for, why you need to call the node driver in this way?

File: Dockerfile-csi-node (hunk 8 above)
Comment (igorgonibm) [reply Dockerfile-csi-node]: yes

File: node/scripts/entrypoint.sh (hunk 9 above)
Comment (igorgonibm) [reply node/scripts/entrypoint.sh]: To be consistent with the way it is done in controller
Istead of starting in ENTRYPOINT directly added script which does it.

Comment (27149chen) [reply node/scripts/entrypoint.sh]: first of all, it is not part of this pr, you should do it in other prs.
And you should also use exec as controller did.

**Pull request merged**